pkg/client/ssa_client.go

```diff
@@ -111,6 +173,32 @@ func (c *SSAApplyClient) ApplyObject(ctx context.Context, obj client.Object, opt
 	return nil
 }
 
+func (c *SSAApplyClient) migrateSSA(ctx context.Context, obj client.Object) error {
+	orig := obj.DeepCopyObject().(client.Object)
+	if err := c.Client.Get(ctx, client.ObjectKeyFromObject(obj), orig); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return fmt.Errorf("failed to get the object from the cluster while migrating managed fields: %w", err)
+		}
+		orig = nil
+	}
+
+	if orig != nil {
+		oldFieldOwner := c.NonSSAFieldOwner
+		if len(oldFieldOwner) == 0 {
+			// this is how the kubernetes api server determines the default owner from the user agent
+			// The default user agent has the form of "name-of-binary/version information etc.".
+			// The owner is the first part of the UA unless explicitly specified in the request URI.
+			oldFieldOwner = strings.Split(rest.DefaultKubernetesUserAgent(), "/")[0]
+		}
```

Comment:
related to my previous comment, do you think that we would need to se the owner name explicitly? Couldn't we rely on the default behavior only?

Reply:
We could, but the default behavior makes assumptions that I was not 100% sure we satisfy everywhere in the codebase, so I wanted to have an explicit option available for the case we find a place where the defaults would not be applicable.

The default behavior assumes that:
1. There is no user agent set explicitly in the config of any the kubernetes client.
2. We don't explicitly set the field owner with any of the existing CRUD operations.

Reply:
>There is no user agent set explicitly in the config of any kubernetes client.
>We don't explicitly set the field owner with any of the existing CRUD operations with the kubernetes clients.

I'm not aware of any place that would change any of these explicitly, so everything should comply. 

Anyway, thanks for the explanation. It's fine to keep it there if not required to be set explicitly by default (only as a backup solution for some corner-cases).